lib/account.rb

```diff
@@ -0,0 +1,106 @@
+module Bank
+
+  class Account
+    @@count_accounts = 0
+
+    attr_accessor :id, :balance, :date, :owner
+
+    def initialize(id, date)
+      @balance = initial_balance
+      @id = id
+      @date = date
+      @@count_accounts += 1
+      @owner = nil
+    end
+
+    def set_owner(owner)
+      @owner = owner
+    end
+
+    #get a row from index and return a new account
+    def self.create_account(person_array)
+      account_id = person_array[0].to_i
+      account_balance = person_array[1].to_i
+      account_date = person_array[2]
+      return Bank::Account.new(account_id, account_balance, account_date)
+    end
+
+    #read the CSV file and row by row use the "creat account" method to create accounts and
+    #push it to an array - and return it
+    def self.all
+      sample = CSV.read("./support/accounts.csv")
+      accounts = []
+      sample.each do |row|
+        account = create_account(row)
+        accounts.push(account)
+      end
+      return accounts
+    end
+
+    def self.find(id)
+      sample = CSV.read("./support/accounts.csv")
+      sample.find do |row|
+        if row[0].to_i == id
+          return create_account(row)
+        end
+      end
+    end
+
+
+
+
+    def initial_balance(min_balance = 0)
+      puts "$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$".center(70)
+      puts
+      puts "$$$$$$$ Welcome to Tammy's Bank! $$$$$$$".center(70)
+      puts "How much $$$ would you like to deposit to your new account ?".center(70)
+      puts
+      puts "$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$".center(70)
+      money = gets.chomp.to_i
+      if money < 0
+        raise ArgumentError.new("You can't deposit negativ amount of money")
+        money = 0
+      elsif  money < min_balance.to_i
+        raise ArgumentError.new("You can't deposit less then $#{min_balance/100}")
+        money = 0
+      end
+      return money
+    end
+
+    def withdraw (amount, fee = 0, min_balance = 0)
```

Comment:
I like that you've used default variables for the "normal" case with no fees or "other" minimum balance